publish_courses/api.py

```diff
@@ -159,171 +147,109 @@ def create(self, request, *args, **kwargs) -> Response:
         account = f'sis_account_id:school:{account}'
         term = f'sis_term_id:{term}'
 
-        canvas_courses = self.get_canvas_courses(account, term)
         if not selected_courses:
-            selected_courses = canvas_courses
+            logger.info(f"No courses selected for {account} and term {term}.")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

You have a log statement that might use unsanitized input. Depending on the context, this could result in:

1. A log injection attack that breaks log integrity, forges log entries, or bypasses monitors that use the logs. To increase the security of your code, sanitize your inputs before logging them. [Learn more](https://cwe.mitre.org/data/definitions/117.html)

2. A sensitive information leak that exposes users&#39; credentials, private information, or identifying information to an attacker. To preserve privacy in your code, redact sensitive user information before logging it. [Learn more](https://cwe.mitre.org/data/definitions/532.html)

Similar issue at line numbers 176, 185, 226, and 252.